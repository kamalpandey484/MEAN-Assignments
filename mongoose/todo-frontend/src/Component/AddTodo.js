import axios from 'axios';
import React, { Component } from 'react';
import '../App.css';
import uuid from 'uuid';
// import { browserHistory } from 'react-router';


class AddTodo extends Component {
    constructor() {
        super();
        this.state = {
            newTodo: '',
            users: []
        };

        // this.apiUrl = 'https://5cc0347182ec6a00149f3971.mockapi.io/users';
        this.apiUrl = 'http://localhost:3001/';

        this.addTodo = this.addTodo.bind(this);
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        this.setState({
            newTodo: event.target.value
        });
    }

    async addTodo(e) {
        e.preventDefault();
        const response = await axios.post(`${this.apiUrl}`, {
            id: uuid(),
            name: this.state.newTodo
        });
        console.log(response);
        const users = this.state.users;
        users.push(response.data);
        this.props.history.push('/')
    }
    render() {
        return (
            <div className="App `col-6 mx-auto col-md-6`">
                <header className="App-header">
                    <h1 className="App-title text-center mt-3">Add Todo</h1>
                </header>
                <div className="container">
                    <form onSubmit={this.addTodo}>
                        <input
                                id="todoInput"
                                type="text"
                                name="todo"
                                className="my-4 form-control"
                                placeholder="Add a new todo"
                                onChange={this.handleChange}
                                value={this.state.newTodo}
                                required
                            />
                            <button
                                className="btn-success mb-3 form-control"
                            >Add todo
                            </button>
                    </form>
                </div>
            </div>
        );
    }
}

export default AddTodo;