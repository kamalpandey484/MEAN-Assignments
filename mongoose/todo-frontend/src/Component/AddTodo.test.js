import React from 'react';
import AddTodo from "./AddTodo";
import { shallow, mount } from "enzyme";
import renderer from 'react-test-renderer';

const wrapper = shallow(<AddTodo />);
const wrapperMount = mount(<AddTodo />);
describe("Testing AddTodo Component!", () => {
    test("Is snapshot Same", () => {
        const tree = renderer.create(<AddTodo />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test("Is AddTodo component rendered", () => {
        expect(wrapper.find(".App").length).toBe(1);
    });
    test("does All fields (input and button) rendered",()=>{
        expect(wrapperMount.containsMatchingElement(<h1>Add Todo</h1>)).toEqual(true);
        expect(wrapperMount.containsMatchingElement(<input name="todo"/>)).toEqual(true);
        expect(wrapperMount.containsMatchingElement(<button>Add todo</button>)).toEqual(true);
    })
    describe('Todo Input', () => {
        it('should respond to change event and change the state of the AddTodo Component', () => {
            wrapper.find('#todoInput').simulate('change', {target: {name: 'todo', value: 'Test More Components!'}});
            expect(wrapper.state('newTodo')).toEqual('Test More Components!');
        })

    })
});