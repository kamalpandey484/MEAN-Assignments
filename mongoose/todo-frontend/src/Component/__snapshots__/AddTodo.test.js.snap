// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing AddTodo Component! Is snapshot Same 1`] = `
<div
  className="App \`col-6 mx-auto col-md-6\`"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title text-center mt-3"
    >
      Add Todo
    </h1>
  </header>
  <div
    className="container"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="my-4 form-control"
        id="todoInput"
        name="todo"
        onChange={[Function]}
        placeholder="Add a new todo"
        required={true}
        type="text"
        value=""
      />
      <button
        className="btn-success mb-3 form-control"
      >
        Add todo
      </button>
    </form>
  </div>
</div>
`;
